frontend/src/routes/marketLink.svelte

```diff
@@ -3,20 +3,69 @@
 	import Button from '$lib/components/ui/button/button.svelte';
 	import { websocket_api } from 'schema-js';
 	import type { Readable } from 'svelte/store';
+	import { writable } from 'svelte/store';
+	import { Star } from 'lucide-svelte';
 
 	export let market: Readable<websocket_api.IMarket>;
 	$: marketIdParam = Number($page.params.id);
 	$: closed = $market.closed;
+
+	const starredStore = writable(
+		localStorage.getItem(`is_starred_${Number($market.id)}`) === 'true'
+	);
+	$: starred = $starredStore;
+
+	function handleStarClick() {
+		starredStore.update((currentValue) => {
+			const newValue = !currentValue;
+			localStorage.setItem(`is_starred_${Number($market.id)}`, newValue ? 'true' : 'false');
+			return newValue;
+		});
+	}
+
+	let isHovering = false;
 </script>
 
-<li class:order-2={!closed} class:order-4={closed}>
+<li
+	class:order-2={!closed && starred}
+	class:order-3={!closed && !starred}
+	class:order-5={closed && starred}
+	class:order-6={closed && !starred}
+	class="flex items-center gap-2"
+>
 	{#if marketIdParam === $market.id}
-		<Button class="inline whitespace-normal text-start text-lg" variant="link" disabled>
+		<button
+			on:click={handleStarClick}
+			on:mouseenter={() => (isHovering = true)}
+			on:mouseleave={() => (isHovering = false)}
+			class="mt-1 inline rounded-full p-1 focus:outline-none"
+			aria-label={starred ? 'Unstar market' : 'Star market'}
+		>
+			<Star
+				color={starred || isHovering ? 'gold' : 'slategray'}
+				fill={starred ? (isHovering ? 'none' : 'gold') : 'none'}
+				size="20"
+			/>
+		</button>
```

Comment:
Please use the Button component, you should be able to add all of the attributes on the component and it'll pass them to the underlying button element. If none of the existing variants work (I think there's variant='icon'), you can add the variant in the file where they're defined.